### PR TITLE
Allow exporting submission for users with at least one correct submission

### PR DESCRIPTION
**Pull request by jorg-vr**
This pull request allows filtering out students who don't have a correct submission for a certain exercise from the result.

image

- [x] Tests were added

Closes https://github.com/dodona-edu/dodona/issues/2561

### CODE REVIEW

**Review by chvp — APPROVED** [review]

File: app/helpers/export_helper.rb [hunk 1]
```diff
@@ -80,7 +80,11 @@ def with_info?
 
     # includes all students in the zip by adding an empty text file for exercises they did not finish
     def all_students?
-      @options[:all_students].present?
+      @options[:filter_students].present? && @options[:filter_students] == 'all'
+    end
+
+    def only_correct_submitted_students?
+      @options[:filter_students].present? && @options[:filter_students] == 'correct'
```
Comment (bmesuere) [inline app/helpers/export_helper.rb:87]: Are the first parts of these checks needed here? (I can be wrong due to my lack of sleep)

File: config/locales/views/exports/nl.yml [hunk 2]
```diff
@@ -40,6 +40,8 @@ nl:
       all_students_help: Studenten die nooit hebben ingediend zullen een lege map hebben in het resultaat.
       only_submitted: Enkel studenten die minstens één keer ingediend hebben
       only_submitted_help: Studenten die nooit hebben ingediend zullen niet te vinden zijn in het resultaat.
+      only_correct_submitted: Enkel studenten die minstens één correcte oplossing hebben ingediend
+      only_correct_submitted_help: Studenten die nooit een correcte oplossing hebben ingediend zullen niet te vinden zijn in het resultaat.
```
Comment (bmesuere) [inline config/locales/views/exports/nl.yml:44]:
```suggestion
      only_submitted_help: Studenten die nooit hebben ingediend zullen niet opgenomen worden in het resultaat.
      only_correct_submitted: Enkel studenten die minstens één correcte oplossing hebben ingediend
      only_correct_submitted_help: Studenten die nooit een correcte oplossing hebben ingediend zullen niet opgenomen worden in het resultaat.
```

File: config/locales/views/exports/en.yml [hunk 3]
```diff
@@ -40,6 +40,8 @@ en:
       all_students_help: Students who never submitted have an empty folder in the result.
       only_submitted: Only students with at least one submission
       only_submitted_help: Students who never submitted will not be found anywhere in the result.
+      only_correct_submitted: Only students with at least one correct submission
+      only_correct_submitted_help: Students who never submitted a correct solution will not be found anywhere in the result.
```
Comment (bmesuere) [inline config/locales/views/exports/en.yml:44]:
```suggestion
      only_submitted_help: Students who never submitted will not be recorded in the result.
      only_correct_submitted: Only students with at least one correct submission
      only_correct_submitted_help: Students who never submitted a correct solution will not be recorded in the result.
```

**Review by bmesuere — CHANGES_REQUESTED** [review]

File: config/locales/views/exports/en.yml (hunk 3 above)
Comment (jorg-vr) [reply config/locales/views/exports/en.yml:44]: Is recorded used like this in English? It feels odd to me, but I could be wrong.
Maybe 'included' is better?

Comment (bmesuere) [reply config/locales/views/exports/en.yml]: included is ok for me

**Review by bmesuere — APPROVED** [review]

**Pull request merged**